packages/flutter/lib/src/material/date_picker.dart

```diff
@@ -672,6 +726,7 @@ class _DatePickerDialogState extends State<_DatePickerDialog> {
           firstDate: widget.firstDate,
           lastDate: widget.lastDate,
           selectableDayPredicate: widget.selectableDayPredicate,
+          onHeaderTap: () => _handleModeChanged(_DatePickerMode.year),
```

Comment:
We only use => for one-liners that return a value, so () { _handleModeChanged(_DatePickerMode.year); },